src/Omega_h_mesh.cpp

```diff
@@ -939,4 +939,24 @@ Adj Mesh::ask_revClass (Int edim) {
   return derived_rc;
 }
 
+template <typename T>
+void Mesh::change_tagToBoundary(Int ent_dim, Int ncomps, std::string const &name) {
+  auto mesh_field = get_array<T>(ent_dim, name);
+  auto boundary_ids = (ask_revClass(ent_dim)).ab2b;
+  auto n_bEnts = boundary_ids.size();
+  Write<T> b_field(n_bEnts*ncomps);
+  auto f = OMEGA_H_LAMBDA(LO i) {
+    auto id = boundary_ids[i];
+    for (LO n = 0; n < ncomps; ++n) {
+      b_field[i*ncomps + n] = mesh_field[id*ncomps + n];
+    }
+  };
+  parallel_for(n_bEnts, f, "get_bdryField");
+  remove_tag(ent_dim, name);
+  std::string new_name = name;
+  new_name.append("_boundary");
+  OMEGA_H_CHECK(!has_tag(ent_dim, name));
+  add_tag<T>(ent_dim, name, ncomps, Read<T>(b_field));
```

Comment:
It looks like the boundary tag/field is being stored as a regular tag.  What happens when the mesh is written to file, or the mesh is modified or repartitioned?

Reply:
Thank you for the comment. 
For file i/o, i was planning to define OMEGA_H_INTERIOR_VALUE flag
and work on a function which compliments change_tagToBoundary 
(something like boundaryTagtoMesh or so), and will use the flag for the entities
which are not on boundary to convert rctag to tag.

The interpretation of this will be properly documented for the user.

similarly during adaptation and migration, convert rctags to tags for solution
transfer to new mesh.
after mesh modification, the rc info will be created again
and then create rctags from tags using the newly created rc info.
the operations will not be cheap, but thats not unexpected.

Reply:
This is an interesting idea.  As you indicated though, I'm concerned about the performance of converting back and forth from 'rcTag' to 'Tag'.  Given that we don't need to support migration, adaptation, and file i/o in the short term, I think we should document that these capabilities are *not* supported and proceed with other work (e.g., use in gitrm, etc.).

Reply:
As discussed in the meeting, we will proceed with the implementation using the 'rcTag' to 'Tag' conversion with support for migration, adaptation, file i/o, visualization etc, as long as the:
- performance of code not using 'rcTag's remains the same, and 
- code changes to support 'rcTag' to 'tag' conversions require minimal changes.

Once the implementation is in place, we can run tests to determine the performance overhead (memory and time) of the tag conversion.